### PR TITLE
Andrew/semester form

### PR DESCRIPTION
**Pull request by amillman**
Semester form, still need to get the current semester in the beginning of app start

### CODE REVIEW

File: sage-ios/SAGE/SAGE/AdminTableViewController.swift [hunk 1]
```diff
@@ -66,7 +66,11 @@ class AdminTableViewController: UITableViewController {
         case 2:
             switch indexPath.row {
             case 0:
-                self.presentViewController(SemesterVerifyViewController(), animated: true, completion: nil)
+                if false { //if semester exists
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminTableViewController.swift]: put a TODO:  here so we can find it later

File: sage-ios/SAGE/SAGE/AdminTableViewController.swift [hunk 2]
```diff
@@ -105,10 +109,17 @@ class AdminTableViewController: UITableViewController {
             }
         case 2:
             if indexPath.row == 0 {
-                cell.textLabel?.text = "End Fall 2015"
-                let icon = FAKIonIcons.logOutIconWithSize(iconSize)
-                    .imageWithSize(CGSizeMake(iconSize, iconSize))
-                cell.imageView?.image = icon
+                if false { // if semester exists
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminTableViewController.swift]: same

File: sage-ios/SAGE/SAGE/EndSemesterViewController.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        self.endView.showFirstLabel()
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        UIApplication.sharedApplication().statusBarStyle = .LightContent
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/EndSemesterViewController.swift:40]: should you be using the setStatusBarStyle method?

File: sage-ios/SAGE/SAGE/Semester.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+enum Term: Int {
+    case Spring = 0
+    case Fall = 1
+}
+
+class Semester: NSObject {
+    
+    var id: Int = -1
+    var startDate: NSDate?
+    var term: Term = .Spring
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/Semester.swift:20]: we should have a default term value for when the data doesnt get passed back or for some other bad reason - like Term.Default

File: sage-ios/SAGE/SAGE/Semester.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            default: break
+            }
+        }
+        super.init()
+    }
+    
+    //
+    // MARK: - Public Methods
+    //
+    func stringFromStartDate() -> NSString {
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/Semester.swift]: specify that this is display string - rename as displayStringFromStartDate or something

File: sage-ios/SAGE/SAGE/SemesterOperations.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            let semesterDict = (data as! [String: AnyObject])["semester"] as! [String: AnyObject]
+            let createdSemester = Semester(propertyDictionary: semesterDict)
+            completion(createdSemester)
+            }) { (operation, error) -> Void in
+                failure("Could not start semester.")
+        }
+    }
+    
+    static func endSemester(completion: () -> Void, failure: (String) -> Void) {
+        // GET CACHED semester
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/SemesterOperations.swift]: does this mean get from keychain???

File: sage-ios/SAGE/SAGE/StartSemesterView.swift [hunk 7]
```diff
@@ -0,0 +1,129 @@
+//
+//  StartSemesterView.swift
+//  SAGE
+//
+//  Created by Andrew Millman on 1/17/16.
+//  Copyright © 2016 Cal Blueprint. All rights reserved.
+//
+
+class StartSemesterView: UIView {
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/StartSemesterView.swift]: inherit from the formview class I made in my last PR! it removes some repeated things
it handles all the scrollview stuff and the layouts for it, parts of setupSubviews, and something else idk

File: sage-ios/SAGE/SAGE/StartSemesterController.swift [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+//  SAGE
+//
+//  Created by Andrew Millman on 1/17/16.
+//  Copyright © 2016 Cal Blueprint. All rights reserved.
+//
+
+import Foundation
+
+class StartSemesterViewController: UIViewController {
+
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/StartSemesterController.swift:12]: inherit from the formcontroller class i made in my last pr, it'll do the finish button and call the completeForm method for you

File: sage-ios/SAGE/SAGE/EndSemesterViewController.swift [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+//  EndSemesterViewController.swift
+//  SAGE
+//
+//  Created by Andrew on 1/1/16.
+//  Copyright © 2016 Cal Blueprint. All rights reserved.
+//
+
+import Foundation
+
+class EndSemesterViewController: UIViewController {
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/EndSemesterViewController.swift:11]: inherit from the formcontroller class i made in my last pr, it'll do the finish button and call the completeForm method for you

File: sage-ios/SAGE/SAGE/EndSemesterViewController.swift (hunk 3 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/EndSemesterViewController.swift:40]: properties should be set this way when possible. only exception is setStatusBarStyle(_:animated:)

Comment (sameerav) [inline sage-ios/SAGE/SAGE/EndSemesterViewController.swift:40]: o ok

File: sage-ios/SAGE/SAGE/Semester.swift (hunk 4 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/Semester.swift:20]: default would be Spring then

File: sage-ios/SAGE/SAGE/SemesterOperations.swift (hunk 6 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/SemesterOperations.swift]: yep

File: sage-ios/SAGE/SAGE/EndSemesterViewController.swift (hunk 9 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/EndSemesterViewController.swift:11]: this is not a form

Comment (sameerav) [inline sage-ios/SAGE/SAGE/EndSemesterViewController.swift:11]: oops

File: sage-ios/SAGE/SAGE/Semester.swift (hunk 4 above)
Comment (sameerav) [inline sage-ios/SAGE/SAGE/Semester.swift:20]: lol i just meant like a way to tell that something went wrong but ok

**Pull request merged**